robots/car_display.h

```diff
@@ -107,6 +107,64 @@ class CarDisplay
         return out;
     }
 
+    std::vector<float> getMouseClickPixelPosition() const
+    {
+        std::vector<float> out;
+        out.push_back(m_MouseClickPosition[0]);
+        out.push_back(m_MouseClickPosition[1]);
+        return out;
+    }
+
+    //! draws the rectangle at the desired coordinates
+    void drawRectangleAtCoordinates(const SDL_Rect &rectangle)
+    {
+        SDL_SetRenderDrawColor(m_Renderer, 0, 0, 255, 255);
+        SDL_RenderFillRect(m_Renderer, &rectangle);
+    }
+
+    //! return the current SDL_Renderer pointer if we want to draw additional graphics
+    //! elements to the screen.
+    SDL_Renderer* getRenderer() {
+        return m_Renderer;
+    }
```

Comment:
If it were me, I'd return a reference rather than a pointer here:
```
SDL_Renderer &getRenderer() {
    return *m_Renderer;
}
```

But it's up to you.